src/crimson/mon/MonClient.h

```diff
@@ -0,0 +1,135 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+
+#pragma once
+
+#include <memory>
+
+#include <seastar/core/future.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/timer.hh>
+
+#include "auth/AuthClientHandler.h"
+#include "auth/AuthMethodList.h"
+#include "auth/KeyRing.h"
+#include "auth/RotatingKeyRing.h"
+
+#include "crimson/net/Connection.h"
+#include "crimson/net/Dispatcher.h"
+
+#include "messages/MAuthReply.h"
+#include "mon/MonMap.h"
+
+#include "mon/MonSub.h"
+
+template<typename Message> using Ref = boost::intrusive_ptr<Message>;
+namespace ceph::net {
+  class Messenger;
+}
+
+struct MMonMap;
+struct MMonSubscribeAck;
+struct MMonGetVersionReply;
+struct MMonCommandAck;
+struct MLogAck;
+struct MConfig;
+
+namespace ceph::mon {
+
+class Connection {
```

Comment:
it looks like Client only needs a forward delcaration of class Connection; consider moving the class itself to MonClient.cc

Reply:
yeah, this is tempting idea. but `pending_conns` is a `std::vector<Connection>`. i will try to make it a `std::vector<std::unique_ptr<Connection>>` in a follow-up PR.

Reply:
vector should work with forward declarations; you probably just need to define `~Client()` in the source file for access to `~Connection()`